sound/soc/sof/ipc4-topology.h

```diff
@@ -364,4 +364,42 @@ struct sof_ipc4_process {
 	u32 init_payload_format;
 };
 
+/**
+ * struct sof_ipc4_pin_format - widgets pin format
+ * @pin_index: input/output pin index for the widget
+ * @iobs: input/output buffer size for input/output pin
+ * @audio_fmt: audio format for the pin
+ *
+ * This structure can be used for both source or sink pin cases,
+ * and the pin_index is relative to the type of pin being used.
+ */
+struct sof_ipc4_pin_format {
+	u32 pin_index;
+	u32 iobs;
+	struct sof_ipc4_audio_format audio_fmt;
+};
+
+/**
+ * struct sof_ipc4_base_module_cfg_ext - base module config extension
+ * @num_sink_pin_fmts: number of sink pin formats in the data member
+ * @num_source_pin_fmts: number of source pin formats in the data member
+ * @reserved: reserved for future use
+ * @priv_param_len: length of optional module specific parameters in the data member
+ * @data: consist of three parts:
+ *        a) num_sink_pin_fmts * sof_ipc4_pin_format followed by
+ *        b) num_source_pin_fmts * sof_ipc4_pin_format, and then followed by
+ *        c) priv_param_len bytes of module specific parameters.
+ */
+struct sof_ipc4_base_module_cfg_ext {
+	/*
+	 * Modules with this structure in their instance initialization payload
+	 * should have one or more sink and source pins
+	 */
+	u16 num_sink_pin_fmts;
+	u16 num_source_pin_fmts;
+	u8 reserved[8];
+	u32 priv_param_len;
+	u8 data[];
```

Comment:
However this part is nowhere to be seen, this is an extension of the base_cfg_ext and if I think about this a bit more, this is going to be really risky business to support in a safe way. There is no indication in anywhere that we have base_cfg_ext or base_cfg_ext_ext sent down to the firmware, right?
If the firmware expects (for some reason) base_cfg_ext_ext but the host sent base_cfg_ext then the firmware will try to interpret random data at offset of `priv_param_len` and if it is not 0 then reads the data as well and if we have even worst luck the expected and random len is matching -> the module will take in some random (or maliciously crafted) data.

This is not the 'safe' fallback way of overriding a struct and the introduced flag for the payload type is also incomplete as we could have a module which needs base_cfg_ext (this is **without** the extended extension) and then we have other modules which likes to have base_cfg_ext_ext.
In any case, we need a flag telling that the MOD_INIT_INSTANCE payload needs to have base_cg_ext, which should not come from a blob.
While I still think this is a wrong thing: a blob could have a flag that it needs to be sent with MOD_INIT_INSTANCE via base_cfg_ext_ext.
Even the non ext_ext type should use this ext_ext struct to 0 out the priv_param_len for safety, but what about other implementation following the documentation (remember: base_cfg_ext_ext is a made-up one here), that is _not_ going to zero the len and the firmware in that case will just going to do some interesting stuff.

Do we expect this base_cfg_ext_ext to be only used with new modules (new UUIDs) and existing modules (with known UUIDs) will never support this?